mesheryctl/pkg/utils/spreadsheet_update.go

```diff
@@ -159,38 +163,72 @@ func updateModelsSheet(srv *sheets.Service, cred, sheetId string, values []*Mode
 	if err != nil {
 		return err
 	}
-	Log.Info("Appending", len(marshalledValues), "in the models sheet")
+	Log.Info("Appending", len(marshalledValues), "in the models sheet") // appendSheet or appendToCSVFile
 	err = appendSheet(srv, cred, sheetId, ModelsSheetAppendRange, marshalledValues)
 
 	return err
 }
 
-func updateComponentsSheet(srv *sheets.Service, cred, sheetId string, values []*ComponentCSV) error {
+func updateComponentsSheet(srv *sheets.Service, cred, sheetId string, values []*ComponentCSV, csvPath string) error {
 	marshalledValues, err := marshalStructToCSValues[ComponentCSV](values)
 	Log.Info("Appending", len(marshalledValues), "in the components sheet")
 	if err != nil {
 		return err
 	}
-	err = appendSheet(srv, cred, sheetId, ComponentsSheetAppendRange, marshalledValues)
+	err = appendSheet(srv, cred, sheetId, ComponentsSheetAppendRange, marshalledValues, csvPath)
 
 	return err
 }
 
-func appendSheet(srv *sheets.Service, cred, sheetId, appendRange string, values [][]interface{}) error {
+func appendSheet(srv *sheets.Service, cred, sheetId, appendRange string, values [][]interface{}, csvPaths ...string) error {
```

Comment:
Please run and confirm using the spreadsheet Id and local csv and check if everything works fine.